prog/dftb+/api/mm/capi.F90

```diff
@@ -475,4 +486,26 @@ function fortranChar(cstring, maxlen)
   end function fortranChar
 
 
+  !> Handles the optional output file name (which should be a NULL-ptr if not present)
+  subroutine handleOutputFileName(outputFileName, outputUnit, tOutputOpened)
+    type(c_ptr), intent(in) :: outputFileName
+    integer, intent(out) :: outputUnit
+    logical, intent(out) :: tOutputOpened
+
+    character(c_char), pointer :: pOutputFileName
+    character(:), allocatable :: fortranFileName
+
+    if (c_associated(outputFileName)) then
+      call c_f_pointer(outputFileName, pOutputFileName)
+      fortranFileName = fortranChar(pOutputFileName)
+      open(newunit=outputUnit, file=fortranFileName, action="write")
```

Comment:
Is it worth trapping the error return from this (as the filename might be garbage)?

Reply:
Not at the moment (currently we do not pass back any errors anway...)